engine/src/main/java/org/terasology/engine/logic/characters/CharacterSystem.java

```diff
@@ -315,35 +315,35 @@ private boolean isPredictionOfEventCorrect(EntityRef character, ActivationReques
 
         Vector3f originPos = location.getWorldPosition(new Vector3f());
         if (!(event.getOrigin().equals(originPos, 0.0001f))) {
-            String msg = "Player {} seems to have cheated: It stated that it performed an action from {} but the predicted position is {}";
-            logger.info(msg, getPlayerNameFromCharacter(character), event.getOrigin(), originPos);
+            logger.info("Player {} seems to have cheated: It stated that it performed an action from {} but the predicted position is {}",
+                    getPlayerNameFromCharacter(character), event.getOrigin(), originPos);
             return false;
         }
 
         if (event.isOwnedEntityUsage()) {
             if (!event.getUsedOwnedEntity().exists()) {
-                String msg = "Denied activation attempt by {} since the used entity does not exist on the authority";
-                logger.info(msg, getPlayerNameFromCharacter(character));
+                logger.info("Denied activation attempt by {} since the used entity does not exist on the authority",
+                        getPlayerNameFromCharacter(character));
```

Comment:
Does our logging framework allow for passing a `Supplier<String>` for the log message? Here, it would be nice to avoid the computation of `getPlayerNameFromCharacter(character)` until we actually have to log it. Maybe that's guarded sufficiently, though, and INFO logs are enabled by default currently 🤷

Reply:
My understanding was that the parametrization does exactly that - avoid the computation until we actually have to log it. But I don't know for sure, so I could be misunderstanding how this works.

Reply:
it is so @jdrueckert :
https://dennis-xlc.gitbooks.io/the-logback-manual/content/en/chapter-2-architecture/logger-appenders-and-layouts/parameterized-logging.html

Reply:
@soloturn the parts mentioned in that manual I understood. It doesn't speak to method calls in the parameters, though.
But @skaldarnar's comment above explains it nicely 👍

Reply:
ha, with you guys one learns something new all the time :) there is indeed a fluent API to slf4j-2.0, which allows to pass suppliers:
https://www.slf4j.org/manual.html#fluent